pkg/publisher/combo/fanout_test.go

```diff
@@ -24,11 +24,12 @@ var uninstalledPublisher = mockPublisher{
 
 func TestFanoutPublisher(t *testing.T) {
 	runTestCases(t, map[string]comboTestCase{
-		"single publsiher":      {NewFanoutPublisher(&healthyPublisher), healthyPublisher},
-		"takes first value":     {NewFanoutPublisher(&healthyPublisher, &sleepyPublisher), healthyPublisher},
-		"waits for installed":   {NewFanoutPublisher(&uninstalledPublisher, &sleepyPublisher), sleepyPublisher},
-		"noone is installed":    {NewFanoutPublisher(&uninstalledPublisher), uninstalledPublisher},
-		"waits for good value":  {NewFanoutPublisher(&errorPublisher, &sleepyPublisher), sleepyPublisher},
-		"returns error for all": {NewFanoutPublisher(&errorPublisher, &errorPublisher), errorPublisher},
+		"single publisher":                 {NewFanoutPublisher(&healthyPublisher), healthyPublisher},
+		"takes first value":                {NewFanoutPublisher(&healthyPublisher, &sleepyPublisher), healthyPublisher},
+		"waits for installed":              {NewFanoutPublisher(&uninstalledPublisher, &sleepyPublisher), sleepyPublisher},
+		"noone is installed":               {NewFanoutPublisher(&uninstalledPublisher), uninstalledPublisher},
+		"waits for good value":             {NewFanoutPublisher(&errorPublisher, &sleepyPublisher), sleepyPublisher},
+		"returns error for all":            {NewFanoutPublisher(&errorPublisher, &errorPublisher), errorPublisher},
+		"waits for highest priority value": {NewPrioritizedFanoutPublisher(&sleepyPublisher, &healthyPublisher), sleepyPublisher},
```

Comment:
Could we add a test that it definitely only waits for the maximum time? So if the max is 2s then it clearly only waits for that length. E.g. something like

```go
"only waits for max time": {NewPrioritizedFanoutPublisher(20 * time.Millisecond, &sleepyPublisher, &healthyPublisher), healthyPublisher}
```

And also a test for what happens if the timeout ends before anyone has returned?